modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsEpics.scala

```diff
@@ -340,6 +343,58 @@ final class TcsEpics(epicsService: CaService) {
       }
     })))
 
+  def waitAGInPosition(timeout: Time): SeqAction[Unit] =
+    EpicsCommand.safe(EitherT(Task.async[TrySeq[Unit]]((f) => {
+
+      val resultGuard = new AtomicInteger(1)
+      val lock = new ReentrantLock()
+      def locked(f: => Unit): Unit = {
+        lock.lock()
+        try {
+          f
+        } finally {
+          lock.unlock()
+        }
+      }
+
+      if (!agInPositionAttr.values.isEmpty && agInPositionAttr.value == 1.0) {
+        f(TrySeq(()).right)
+      } else {
+
+        val timer = new Timer
+        val statusListener = new CaAttributeListener[java.lang.Double] {
+
+          override def onValueChange(newVals: util.List[java.lang.Double]): Unit = {
+            if (!newVals.isEmpty && newVals.get(0) == 1.0 && resultGuard.getAndDecrement() == 1) {
+              locked {
+                agInPositionAttr.removeListener(this)
+                timer.cancel()
+              }
+
+              f(TrySeq(()).right)
+            }
+          }
+
+          override def onValidityChange(newValidity: Boolean): Unit = {}
+        }
+
+        locked {
```

Comment:
Uff, I replicated the code used to wait for the TCS in-position flag. I wrote that like two years ago. I'll need some time to remember how this works.